packages/calendar-time-utils/src/formats.ts

```diff
@@ -0,0 +1,69 @@
+// Reference: https://momentjs.com/docs/#/displaying/format/
+export enum FormatToken {
+  D = 'D', // NUMERIC_SINGLE_DAY
+  DD = 'DD', // NUMERIC_TWO_DIGITS_DAY
+  M = 'M', // NUMERIC_SINGLE_MONTH
+  MM = 'MM', // NUMERIC_TWO_DIGIT_MONTH
+  YY = 'YY', // TWO_DIGITS_YEAR
+  YYYY = 'YYYY', // FOUR_DIGITS_YEAR
+  h = 'h', // HOUR_12_SINGLE
+  hh = 'hh', // HOUR_12_TWO_DIGITS
+  H = 'H', // HOUR_24_SINGLE
+  HH = 'HH', // HOUR_24_TWO_DIGITS
+  m = 'm', // MINUTE_SINGLE
+  mm = 'mm', // MINUTE_TWO_DIGITS
+  A = 'A', // PERIOD - https://unicode.org/reports/tr35/tr35-6.html#Date_Format_Patterns
+}
+
+export type TLocaleDateFormatMapping = {
+  [key in FormatToken]?: string;
+};
+
+export type TLocalizedDateFormatMapping = {
+  [locale: string]: TLocaleDateFormatMapping;
+};
+
+// Here is where we translate from the standard characters used in the pattern
+// (calculated in the moment.js library) to the characters used in the locale
+// We just need to provide the translation for the characters that are different
+// For instance, in english we don't need to translate date characters as
+// they are the same as the pattern characters
+export const DATE_FORMAT_LOCALIZED_MAPPINGS: TLocalizedDateFormatMapping = {
```

Comment:
This is where we would need to work if we start supporting new languages.
We just map the pattern characters to localized characters (if needed).